GMG/Scenes/Home/HomeView.swift

```diff
@@ -313,64 +143,51 @@ extension HomeView {
                 if isEditable {
                     endRename(commit: true)
                 } else {
-                    if isSelected {
-                        action()
-                    } else {
-                        isSelected.toggle()
-                    }
+                    tapAction()
                 }
             }
-            .padding(.bottom, isSelected && isMove ? 60 : 0)
         }
         
+
         // MARK: - Rename Helpers
         private func startRename() {
             tempTitle = score.title
             isEditable = true
-            DispatchQueue.main.async {
+            Task { @MainActor in
                 isTitleFocused = true
             }
         }
-        
+
         private func endRename(commit: Bool) {
             if commit {
-                let new = tempTitle.trimmingCharacters(in: .whitespacesAndNewlines)
+                let newTitle = tempTitle.trimmingCharacters(in: .whitespacesAndNewlines)
                 // 빈 문자열로 저장되는 것 방지 + 변경 시에만 저장
-                if !new.isEmpty, new != score.title {
-                    score.title = new
-                    try? modelContext.save()
+                if !newTitle.isEmpty, newTitle != score.title {
+                    renameScoreAction(newTitle)
                 }
             }
             isTitleFocused = false
             isEditable = false
         }
-        
+
+        // MARK: - Color Helpers
         private func colorForIndex(_ i: Int) -> Color {
-            let palette: [Color] = [
-                Color.blue3, Color.blue4, Color.blue5, Color.blue1, Color.blue2,
-            ]
-            let safe = max(i, 1)
-            return palette[(safe - 1) % palette.count]
+            let palette: [Color] = [.blue3, .blue4, .blue5, .blue1, .blue2]
+            return palette[i % palette.count]
         }
```

Comment:
메서드 대신 프로퍼티로 하는 것도 괜찮을 것 같아요~

```suggestion
        private var backgroundColor: Color {
            let palette: [Color] = [.blue3, .blue4, .blue5, .blue1, .blue2]
            return palette[index % palette.count]
        }
```